### PR TITLE
MAIL function

### PR DESCRIPTION
**Pull request by lekoOwO**

### CODE REVIEW

File: server/api.js [hunk 1]
```diff
@@ -247,6 +271,54 @@ async function start() {
...
+                else {
+                    ctx.body = await mailtasks[id]["main"];
+                }
+            }
+        })
+        .post('/mail', async ctx => {
+            const sourceId = ctx.query.source;
+            const source = await db.source.get({ id: sourceId }, pool);
+
+            const mail = (await readFile('email.html')).replace(/\{\{source\}\}/g, source[0].name);
```
Comment (seadog007) [inline server/api.js]: use `§` to simplify this

File: server/api.js [hunk 2]
```diff
@@ -247,6 +271,54 @@ async function start() {
...
+
+            let uuid = uuidv1();
+            while (uuid in mailtasks) {
+                uuid = uuidv1();
+            }
+
+            mailtasks[uuid] = {promises: []};
+            mailtasks[uuid]["main"] = MakeQuerablePromise(new Promise((resolve, reject) => {
+                for (const user of users) {
+                    const mail_inner = mail.replace(/\{\{user\}\}/g, user.name);
```
Comment (seadog007) [inline server/api.js]: Also this

File: server/email.html [hunk 3]
```diff
@@ -0,0 +1,14 @@
+<html>
+    <head></head>
+    <body>
+        <h3>{{user}} 君 您好:</h3>
+        <p>
+            &nbsp;&nbsp;&nbsp;&nbsp;我們是 <a href="https://breach.tw">breach.tw</a> 台灣抓漏小天使團隊。<br>
+            在最近的資料庫更新中，我們發現您的使用者資訊已被<font color="#FF0000">{{source}} </font>洩漏，太慘了：（<br>
+            提醒您，平時應做好基本的資安防護措施，並慎選個資提供對象。
+            若需要法律協助，可諮詢<a href="https://www.laf.org.tw">法律扶助基金會</a>。
+            請節哀順變，並祝您仕途順利、學業進步、事事順心。
```
Comment (seadog007) [inline server/email.html]: ._.

**Review by seadog007 — CHANGES_REQUESTED** [review]

File: server/email.html (hunk 3 above)
Comment (lekoOwO) [reply server/email.html]: Should be good

Comment (seadog007) [reply server/email.html]: ._. 節哀順變啦

Comment (lekoOwO) [reply server/email.html]: 那請節哀可以ㄇ

Comment (seadog007) [reply server/email.html]: 不行

File: server/email.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+                reject(error);
+            } else {
+                resolve(info.response);
+            }
+          });
+    })
+}
+
+module.exports = send;
+
```
Comment (seadog007) [inline server/email.js]: WTF is the empty lines below

**Review by seadog007 — CHANGES_REQUESTED** [review]

File: server/email.js (hunk 4 above)
Comment (lekoOwO) [reply server/email.js]: 留天地阿

Comment (lekoOwO) [reply server/email.js]: 不留天地會出血

Comment (seadog007) [reply server/email.js]: 阿你的天勒

Comment (seadog007) [reply server/email.js]: 阿你的天勒

Comment (lekoOwO) [reply server/email.js]: 等等補上

**Comment by seadog007** [comment]
Use Object for replacing template for future extension.

File: server/db.js [hunk 5]
```diff
@@ -187,9 +187,23 @@ item.add = async function (id, name, abbr, con = null) {
     return insertId;
 }
 
+let mail = async function (source, con = null) {
+    if (!con) {
+        con = await connect();
+    }
+
+    let result = await con.query("SELECT `email` FROM `subscribers` WHERE `hash` IN (SELECT `hash` FROM `breach_log` WHERE `source` = ?) AND email_verify = 1", source)
```
Comment (seadog007) [inline server/db.js]: Oh, FXXX
I forgot to `disabled = 0`

Comment (seadog007) [reply server/db.js]: And also email_verify should be in ``

**Review by seadog007 — CHANGES_REQUESTED** [review]

File: server/db.js (hunk 5 above)
Comment (koru1130) [reply server/db.js]: > 
> 
> Oh, FXXX
> I forgot to `disabled = 0`

Should be "Fxxx" here

**Review by seadog007 — APPROVED** [review]

**Comment by seadog007** [comment]
Seems good without test, wait for Docker CI and merge

**Pull request merged**